gprofiler/client.py

```diff
@@ -78,7 +78,11 @@ def _send_request(
             opts["headers"]["Content-type"] = "application/json"
             buffer = BytesIO()
             with gzip.open(buffer, mode="wt", encoding="utf-8") as gzip_file:
-                json.dump(data, gzip_file, ensure_ascii=False)  # type: ignore
+                try:
+                    json.dump(data, gzip_file, ensure_ascii=False)  # type: ignore
+                except TypeError:
+                    logger.exception("Given data is not a valid JSON!", bad_json=str(data))
```

Comment:
Why do you need this? How can it happen with our structured data? And logging `str(dta)` is probably too much (requests can be in the order of MBs, don't log them all)

Reply:
It's useful mainly for development to have a better and more indicative error message, I don't expect it to happen in production.
I've added a cutoff

Reply:
k. since it confused me twice already, add a comment? Usually when you catch exceptions, it's because you expect them to happen. If you don't expect, comment that

Reply:
Added a comment